src/wp-includes/comment-template.php

```diff
@@ -1889,16 +1889,24 @@ function post_reply_link( $args = array(), $post = null ) {
  *
  * @since 2.7.0
  *
- * @param string $text Optional. Text to display for cancel reply link. If empty,
- *                     defaults to 'Click here to cancel reply'. Default empty.
+ * @param string $text    Optional. Text to display for cancel reply link. If empty,
+ *                        defaults to 'Click here to cancel reply'. Default empty.
+ * @param int    $post_id Optional. The post ID the comment thread is being
+ *                        displayed for. Default current global post ID.
  * @return string
  */
-function get_cancel_comment_reply_link( $text = '' ) {
+function get_cancel_comment_reply_link( $text = '', $post_id = 0 ) {
 	if ( empty( $text ) ) {
 		$text = __( 'Click here to cancel reply.' );
 	}
 
-	$style = isset( $_GET['replytocom'] ) ? '' : ' style="display:none;"';
+	if ( 0 === $post_id ) {
+		$post_id = get_the_ID();
+	}
+
+	$reply_to_id = _get_comment_reply_id( $post_id );
+
+	$style = ! empty( $reply_to_id ) ? '' : ' style="display:none;"';
```

Comment:
Though technically fine, using a conditional check against `0` would be slightly more performant (see https://3v4l.org/b15cS#v7.4.26) and more readable.

Why? 

`_get_comment_reply_id()` will always return an integer data type and `0` means there's no reply-to ID. `empty()` on the other hand is a falsey check for various scalar data types including integer, string, array, float.

```suggestion
	$style = 0 !== $reply_to_id ? '' : ' style="display:none;"';
```

As a side note, the function could be put inline for the conditional check to eliminate the variable assignment. The function's name is clear about what gets returned, thereby making the variable assignment unnecessary. The variable assignment pattern exists in `get_comment_id_fields()` and `comment_form_title().` For consistency, keeping the variable assignment makes sense.